src/photos/ducks/timeline-clusters/index.js

```diff
@@ -36,39 +38,122 @@ const TIMELINE_MUTATIONS = query => ({
     })
 })
 
-const getPhotosByMonth = photos => {
-  let sections = {}
-  photos.forEach(p => {
+const formatDate = date => {
+  return format(date, 'DD MMMM YYYY')
+}
+
+const sectionTitle = album => {
+  if (album.period) {
+    const startPeriod = new Date(album.period.start)
+    const endPeriod = new Date(album.period.end)
+    if (differenceInCalendarDays(endPeriod, startPeriod) > 0) {
+      // TODO Better period display
+      return formatDate(startPeriod) + ' - ' + formatDate(endPeriod)
+    }
+  }
+  return formatDate(new Date(album.name))
+}
+
+/**
+ * A section matches if:
+ * - The datetime is inside the period.
+ * - The section has no period (not a cluster) and is the datetime's day.
+ */
+const matchingSection = (sections, datetime) => {
+  return Object.keys(sections).find(date => {
+    if (sections[date].period) {
+      const startPeriod = new Date(sections[date].period.start)
+      const endPeriod = new Date(sections[date].period.end)
+      return datetime >= startPeriod && datetime <= endPeriod
+    } else {
+      // If the section has no period, it is not a cluster but a daily section
+      return differenceInCalendarDays(datetime, new Date(date)) === 0
+    }
+  })
+}
+
+/**
+ * Save the photos not clustered in the sections.
+ * It simulates a clustering, waiting for the actual one to be processed.
+ */
+const handlePhotosNotClustered = (sections, photosNotClustered) => {
+  photosNotClustered.forEach(photo => {
     const datetime =
-      p.metadata && p.metadata.datetime ? p.metadata.datetime : Date.now()
-    // here we want to get an object whose keys are months in a l10able format
-    // so we only keep the year and month part of the date
-    const month = datetime.slice(0, 7) + '-01T00:00'
-    /* istanbul ignore else */
-    if (!sections.hasOwnProperty(month)) {
-      sections[month] = []
+      photo.metadata && photo.metadata.datetime
```

Comment:
To avoid this kind of conditions, we can use `get(photo, 'metadata.datetime', photo.created_at)` from lodash ;)

Reply:
I'm not a big fan of using this kind of syntaxic help from external libs, but I might be too old fashion ;)

Reply:
Yeah no big deal, I just find it gets more readable when you have more than one test like that. But the code is fine as it is :)